code/modules/research/rdconsole.dm

```diff
@@ -431,6 +431,10 @@ proc/CallMaterialName(ID)
 
 					var/P = being_built.build_path //lets save these values before the spawn() just in case. Nobody likes runtimes.
 					var/O = being_built.locked
+					
+					if(ispath(P, /obj/item/weapon/stock_parts))

```

Comment:
Designs already have a `construction_time` var - Make the console use that var, then set this var lower for the stock parts.